src/Ledger/Ratify.lagda

```diff
@@ -151,8 +251,48 @@ module _ (Γ : RatifyEnv) (cc : CCData) (votes : (GovRole × Credential) ⇀ Vot
   actualVotes : VDeleg ⇀ Vote
   actualVotes = mapKeys (credVoter CC) actualCCVotes (λ where _ _ refl → refl)
               ∪ᵐˡ actualPDRepVotes ∪ᵐˡ actualDRepVotes
-              ∪ᵐˡ mapKeys (uncurry credVoter) votes (λ where _ _ refl → refl) -- TODO: make `actualVotes` for SPO
-
+              ∪ᵐˡ mapKeys (uncurry credVoter) votes (λ where _ _ refl → refl)
+              -- TODO: make `actualVotes` for SPO
+\end{code}
+} %% end small
+\caption{%Ratify i:
+Types and proofs for the ratification of governance actions}
+\label{fig:defs:ratify-i}
+\end{figure*}
+The code in Figure~\ref{fig:defs:ratify-i} defines some of the types required for ratification of a governance action.
```

Comment:
We're on the same page 👍🏼 (see previous comment)

Reply:
By just using `[h!]` instead of `[h]` for the figures, the figure in question is moved up to page 32 (from 35).  I've also been trying to make sure the prose concerning a given figure appears *after* the code that it describes in the lagda file;  that way the code and the explanations have a better chance of ending up in close proximity... but as long as we're placing all non-hidden Agda code inside figures (which is generally a good idea, I think), we are at the mercy of tex's formatting decisions. 😅